presto-kafka/src/main/java/io/prestosql/plugin/kafka/KafkaFieldValueProvider.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka;
+
+import com.google.common.primitives.Shorts;
+import com.google.common.primitives.SignedBytes;
+import io.airlift.slice.Slice;
+import io.prestosql.plugin.kafka.encoder.EncoderColumnHandle;
+import io.prestosql.plugin.kafka.encoder.EncoderFieldValueProvider;
+import io.prestosql.spi.block.Block;
+import io.prestosql.spi.type.CharType;
+import io.prestosql.spi.type.Type;
+import io.prestosql.spi.type.Varchars;
+
+import java.nio.ByteBuffer;
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static io.prestosql.spi.type.Chars.padSpaces;
+import static java.lang.Float.intBitsToFloat;
+import static java.lang.Math.toIntExact;
+import static java.util.Objects.requireNonNull;
+
+/**
+ *
+ */
+public class KafkaFieldValueProvider
+        implements EncoderFieldValueProvider
+{
+    private final EncoderColumnHandle columnHandle;
+    private final Block block;
+    private final int position;
+    private final Type type;
+
+    public KafkaFieldValueProvider(
+            EncoderColumnHandle columnHandle,
+            Block block,
+            int position)
+    {
+        this.columnHandle = requireNonNull(columnHandle, "columnHandle is null");
+        this.block = requireNonNull(block, "block is null");
+        this.position = position;
+        this.type = columnHandle.getType();
+    }
+
+    @Override
+    public EncoderColumnHandle getColumn()
+    {
+        return columnHandle;
+    }
+
+    @Override
+    public boolean getBoolean()
+    {
+        return type.getBoolean(block, position);
+    }
+
+    @Override
+    public long getLong()
+    {
+        return type.getLong(block, position);
+    }
+
+    @Override
+    public int getInt()
+    {
+        return toIntExact(type.getLong(block, position));
+    }
+
+    @Override
+    public short getShort()
+    {
+        return Shorts.checkedCast(type.getLong(block, position));
+    }
+
+    @Override
+    public byte getByte()
+    {
+        return SignedBytes.checkedCast(type.getLong(block, position));
+    }
+
+    @Override
+    public double getDouble()
+    {
+        return type.getDouble(block, position);
+    }
+
+    @Override
+    public float getReal()
```

Comment:
Rethink naming in this class. It's called 'real' after `REA` SQL type (otherwise you would call it `getFloat`).
But `getShort` is called after `short` -- java representation for the `smallint` SQL type.

is it intentional? 
What does this class represent actually?

Reply:
My head was spinning dealing with the data types, I'll revisit the naming choices